Tests/test_delegate2.py

```diff
@@ -0,0 +1,887 @@
+#####################################################################################
```

Comment:
We should probably change the test runner to handle duplicate name (maybe append the dir structure to the name). I'm pretty sure this isn't the only place this occurs.